File: src/main/java/br/com/sergioluigi/personalfinancialcontrol/domain/user/infra/repository/UserRepository.java
package br.com.sergioluigi.personalfinancialcontrol.domain.user.infra.repository;

import br.com.sergioluigi.personalfinancialcontrol.domain.user.infra.repository.entity.UserEntity;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;
import java.util.UUID;

public interface UserRepository extends JpaRepository<UserEntity, UUID> {
    Optional<UserEntity> findByUsername(String username);
}

File: src/main/kotlin/com/sergioluigi/personalfinancialcontrol/adapter/database/repository/UserRepository.kt
package com.sergioluigi.personalfinancialcontrol.adapter.database.repository

import com.sergioluigi.personalfinancialcontrol.adapter.database.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.JpaSpecificationExecutor
import java.util.*

interface UserRepository: JpaRepository<User, UUID>, JpaSpecificationExecutor<User> {
	fun findByCpf(cpf: String): User?
}